### ISSUE TITLE
LEDAPS Installation on Ubuntu GNU/Linux

### ISSUE THREAD

**Issue by safalabolo** [USGS-EROS/espa-surface-reflectance#63]
Dear Ronald,
it would be possible to offer greater signs of "step by step" type for the installation of LEDAPS packages.
I wish I could derive from my pc everything that could derive from the system WEB "Landsat Surface Reflectance High Level Data Products".
Unfortunately, the directions given are very rough, above all for those who like it is not user UNIX / UBUNTU.
Thanks in advance for your reply

**Comment by gschmidt-usgs** [comment]
We expect that the user has some basic Linux knowledge for installation and
building of this package.  We also understand that the need to have various
other libraries, such as the HDF, GeoTIFF, etc. in order to build the
espa-product-formatter may be a challenge.  Can you give us an idea of
where things aren't working out for you or where the instructions may not
be overly clear?  Or is this just a general statement.

We are also working on using Docker to make building of these packages much
easier.

-----------------------------------------------------------------------------------------------------
Gail Schmidt
Software Engineer
Stinger Ghaffarian Technologies (SGT)
Contractor to the U.S. Geological Survey (USGS)
Earth Resources Observation and Science (EROS) Center
47914 252nd Street
Sioux Falls, SD 57198
Phone: 605-593-2272
gail.schmidt.ctr@usgs.gov

NOTICE:  This email may contain confidential, proprietary, or competition
sensitive bid or proposal procurement information.  Unauthorized disclosure
of this information may carry criminal penalties as set forth in the
Procurement Integrity Act, 41 U.S.C. 423, as amended.  Further, the
unauthorized disclosure of certain commercial information by civil servants
may result in fines or imprisonment under the Trade Secrets Act (18 U.S.C.
1905).  If you have received this information in error, please delete it,
including all copies, and notify the sender of the error immediately.

**Comment by yannforget** [comment]
You can use [Anaconda](https://www.continuum.io/why-anaconda) for that. A lot of dependencies are packaged (hdf4, hdf5, netcdf, gdal...). Since it seems that's it could help some people, I just uploaded my conda packages for libgeotiff, hdfeos2, espa-product-formatter, ledaps, lasrc, and cfmask ([On Github](https://github.com/yannforget/conda-recipes), [On Anaconda Cloud](https://anaconda.org/yannforget)).

Basically, once you have Anaconda installed, you can just run:

``` bash
conda config --add channels conda-forge
conda config --add channels yannforget
conda install ledaps lasrc cfmask
```

**Comment by safalabolo** [comment]
Dear Yann,
in order to take account also of the ancillary data, it is right the following sequence of instructions:

conda config --add channels conda-forge
conda config --add channels yannforget
conda install ledaps 
export LEDAPS_AUX_DIR="directory_saved_auxiliary_files"
conda install lasrc 
conda install cfmask

Thanks in advance for your reply.
All the best.

**Comment by safalabolo** [comment]
[log.txt](https://github.com/USGS-EROS/espa-surface-reflectance/files/368723/log.txt)

**Comment by safalabolo** [comment]
Dear Yann,
I currently have a problem with the following library:
libmfhdf.so.0.
I can not fix this problem
and continue to test the entire installation.
Enclose the process log files
thanks in advance for your support and for your response
[log.txt](https://github.com/USGS-EROS/espa-surface-reflectance/files/371079/log.txt)

**Comment by yannforget** [comment]
@safalabolo ,  
`libmfhdf.so` is part of the `hdf4` package. It's a dependency of `hdfeos2` so it supposed to be located in `<conda_install_path>/lib`. Try `conda install -c conda-forge hdf4` if it is not the case.  

These recipes are not supported by USGS-EROS so I think it's better to continue the support elsewhere. [Here](https://github.com/yannforget/conda-recipes/issues) or by mail.

**Comment by palmoreck** [comment]
Use this if you want: 

https://github.com/CONABIO/LEDAPS/blob/master/ledaps_cc/Dockerfile

is something like step by step

In that Dockerfile the ENV command change to export command in a bash shell and the RUN is just execute the command that follows RUN, you can ommit VOLUME and WORKDIR commands

**Comment by safalabolo** [comment]
Dear Palmoreck
thank you very much for your support.
I will try to install and run the whole thing and let you know, if I met problems.
All the best

**Comment by GreatEmerald** [comment]
Is there any particular reason for there not being a DEB package for this? It has a lot of dependencies, sure, but it doesn't seem particularly difficult to package per se. Or is it just that nobody has gone around to doing that yet?

**Comment by ronaldddilley** [comment]
At this time we do not provide DEB or RPM packages for any of our applications.
You can however order products from this site (https://espa.cr.usgs.gov/login) free of charge.

**Comment by GreatEmerald** [comment]
Yes, I am aware of that, but at least at the moment the order queue seems to be quite full. I managed to compile, install and run LEDAPS in a few hours, whereas my order (for a single tile) is still stuck in the queue. So at least for the time being it's more efficient to use LEDAPS.

But my question was why it's not packaged; is there a legal reason, lack of manpower, etc.? Could someone not related to the project just make a PPA or such and package it themselves?

**Comment by ronaldddilley** [comment]
We appreciate your suggestions, and ask for your patience as we pass it on to our government customers to determine what level of service they are able to provide in fulfilling it.

**Comment by daviddemeij** [comment]
> We are also working on using Docker to make building of these packages much
> easier.
> 
> Gail Schmidt

@gschmidt-usgs is this docker image already available somewhere? I did find on the Google Earth Engine page that a docker image was provided to create their Landsat 8 Surface Reflectance product. 
> This product is generated by Google using a Docker image supplied by USGS.
https://gee.stac.cloud/BJmBzK1uPSS1qPWphBuPHgfbcqSkzxvCWZ94q?t=bands
